### ISSUE TITLE
How to disable Checkpointing for Full tuning or PEFT runs?

### ISSUE THREAD

**Issue by premmotgi** [pytorch/torchtune#1487]
I am trying to run single GPU to multinode distributed fine tuning for Llama3-70B and Llama3 8B Models.

Below is my training configuration:
SFT (Llama3 8B & 70B)

Epochs: 3
Gradient Accumulation Steps: 1
Batch Size: 32
Data Type: bf16
Enable Activation Checkpointing: True
Memory Efficient FSDP Wrap: True
FSDP: Enabled
LoRA (Llama3 8B & 70B)

Epochs: 3
Gradient Accumulation Steps: 1
Batch Size: 32
Data Type: bf16
Enable Activation Checkpointing: True
Learning Rate: 3e-4
FSDP: Enabled

Obervation/Error:

I see that after each epoch the tune module is saving checkpoint to NFS share. It takes around 13 minutes on an average after each epoch before the next epoch starts. Due to this I am not able to capture accurate training time for various runs.

Is there any way I could disable checkpointing at all?

**Comment by SalmanMohammadi** [comment]
Hey @premmotgi. You can use the `save_adapter_weights_only` config entry (i.e. `save_adapter_weights_only=True` from the CLI) to only save adapter weights when checkpointing, which should be a fraction of the size. 

Saving to an NFS share may also be a lot slower than a local drive - are you getting good transfer speeds to your NFS share?

**Comment by premmotgi** [comment]
@SalmanMohammadi I understood the use of save_adapter_weights_only = True. does this apply only to LoRa or applies to full tuning as well?

Also, I still think it's worth to have an enabler/disabler for checkpointing as in development phase not everyone would want to save tuned model as it takes space after every epoch. Or maybe have a checkpointing interval config so that checkpointing can be done based on that config.

Saving to an NFS share may also be a lot slower than a local drive - are you getting good transfer speeds to your NFS share? 
- For multinode runs we found it easier to rely on network to store distributed checkpoints to one NFS share as it would help us having tuned model .pt files at one location.

**Comment by felipemello1** [comment]
>Also, I still think it's worth to have an enabler/disabler for checkpointing as in development phase not everyone would want to save tuned model as it takes space after every epoch. Or maybe have a checkpointing interval config so that checkpointing can be done based on that config.

we currently dont have a flag for it, but i share your frustration. Usually i go to the recipe and just comment out the line. If you are interested in contributing, we could add a flag to control the checkpoint.

If do you add the flag, here is a script to update all configs in bulk to include the flag: https://gist.github.com/felipemello1/5f2002433c6da3a21f33d6cdf82e702a

pytorch also has a functionality to save the checkpoint async, but we havent explored it, afaik.

**Comment by premmotgi** [comment]
@felipemello1 No worries at all. I just figured that it would be useful to many in the experimental case of finetuning larger models such as 70b and want to save some storage space. I will add a flag and update the configs in bulk. thanks for sharing the script.

**Comment by cchung85** [comment]
...
> Thanks for raising the issue [@premmotgi](https://github.com/premmotgi). We currently dont have a flag for it, but i share your frustration. Usually i go to the recipe and just comment out the line. If you are interested in contributing, we could add a flag to control the checkpoint.
> 


Could you tell us which line(s) to comment out? 

Without commenting out, I got this Error in tune run ...: /tmp/output/checkpoint/config.json not found! 
(This is odd and totally unexpected... Checkpoint dir should not be used upon startup.)

I am using mistral 7B, and fine tune using lora_finetune_single_device template.  Tried both Lora and qLora:
tune cp mistral/7B_qlora_single_device   mistral-7B_qlora_single_device
tune cp mistral/7B_qora_single_device   mistral-7B_lora_single_device

The relevant yaml entry related to checkpoint are the same in both lora and qlora:

model:
   _component_: torchtune.models.mistral.qlora_mistral_7b # or .lora_mistral_7b
  ...
checkpointer: 
  _component_: torchtune.training.FullModelHFCheckpointer
  checkpoint_dir: /tmp/output/checkpoint
  checkpoint_files: [
    pytorch_model-00001-of-00002.bin,
    pytorch_model-00002-of-00002.bin
  ]
  recipe_checkpoint: null
  output_dir: ${output_dir}
  model_type: MISTRAL
  resume_from_checkpoint: False
  #save_adapter_weights_only: False

resume_from_checkpoint: False
save_adapter_weights_only: False

**Comment by felipemello1** [comment]
can you try just downloading the model again?

```
tune download mistralai/Mistral-7B-v0.1 --output-dir /tmp/Mistral-7B-v0.1 --ignore-patterns "*.safetensors"
```

notice that the command will download the model to "/tmp/Mistral-7B-v0.1"

And when we try to load the model, we will load from there:
```
checkpointer:
  _component_: torchtune.training.FullModelHFCheckpointer
  checkpoint_dir: /tmp/Mistral-7B-v0.1
```

In your config, you are trying to load the model from "checkpoint_dir: /tmp/output/checkpoint", so you need to make sure that the model contents are there, e.g. the config.json.

In other words, this doesnt seem to be a problem with **saving** the checkpoint. This is a problem with **loading** the checkpoint, aka the trained model.

PS: I wouldnt recommend using mistral. We only support its older version v0.1.  The newer versions have different architectures and we dont support them in tune yet. Good alternatives are the qwen and llama models.
